weave-js/src/components/PagePanelComponents/Home/Browse3/pages/CallsPage/CallsTable.tsx

```diff
@@ -191,6 +191,9 @@ export const CallsTable: FC<{
 
   // Fetch the calls
   const calls = useCallsForQuery(entity, project, effectiveFilter);
+  if (calls.refetch) {
+    baseContext.refetchCalls = calls.refetch;
```

Comment:
This pattern would not be ideal in situations where there are 2 calls table on the screen at the same time (as there is when you open an eval). In general, it is not good practice to assign some state like this to a global symbol. Two comments:
1. If you wanted to do a pattern like this, you would want to have a react context provider somewhere above that allows children to "register" and "unregister" refetch triggers.
2. But even that is a little odd - and you might want to reach for a reducer pattern with listeners. 

Still, all these options are a bit odd - you really want someway to "broadcast" changes.

Let's see... in the best case scenario, upon deletion, two events occur:
1. All components on the page which include the deleted element adjust
2. If the primary view is of the deleted element, we navigate away.

Furthermore, when we have renaming, we want similar behavior: all cases of the name change are reflected immediately. 

I can think of 4 families of solutions:
1. Don't do anything and just let the next user-invoked navigation reflect the changes.
2. Hard refresh the whole page.
3. Broadcast change events to listening components to manually reflect the change.
    * Note: this is closest to your current implementation (where "reflect" is facilitated via a refresh) 
4. Automatically update any query hooks with the updated information.
    * The challenge here is not to be overly aggressive causing full-page reloads. Might be widespread refectors
5. are there others?...

What i don't like about #3 is that we put the responsibility on every rendering component to "listen" listen for changes. This is a lot a coupling in the design.

#4 is probably the best option across the board if done correctly. How does this relate to polling?

I wonder if the same mechanism which is hypothetically triggered in a poll event should be triggered after a mutation event? Where such a mechanism is: "refresh all active queries, without re-triggering loading/empty states". Yeah, doing this correctly might require a more widespread refactor.

Ok, so options...:
1. (low-cost version of #4) I wonder if a low-cost change to `OptionalTraceServerClientContextProvider` that allows mutation hooks to refresh the client would be jarring? Might be worth a try.
2. Do what you are doing now, but with a proper context
3. Somehow have the other call hooks react to the delete event

Reply:
brilliant comment, will work through these and have a soln by tue